src/helpers/render-document-file.js

```diff
@@ -403,7 +622,33 @@ async function renderDocumentFile(docxDocumentInstance, properties = {}) {
 
   const xmlFragment = fragment({ namespaceAlias: { w: namespaces.w } });
 
-  const populatedXmlFragment = await convertVTreeToXML(docxDocumentInstance, vTree, xmlFragment);
+  const populatedXmlFragment = await convertVTreeToXML(
+    docxDocumentInstance,
+    vTree,
+    xmlFragment,
+    imageOptions
+  );
+
+  // Log cache statistics at the end of document generation
+  const cacheStats = getImageCacheStats();
+  if (
+    (cacheStats.size > 0 || cacheStats.retryStats.totalAttempts > 0) &&
+    imageOptions.verboseLogging
+  ) {
+    // eslint-disable-next-line no-console
+    console.log(`[CACHE] Image processing statistics:`, {
+      totalImages: cacheStats.size,
+      successful: cacheStats.successCount,
+      failed: cacheStats.failureCount,
+      cacheHitRatio:
+        cacheStats.size > 1 ? 'Cache prevented duplicate downloads' : 'No duplicates found',
+      retries: {
+        totalAttempts: cacheStats.retryStats.totalAttempts,
+        successAfterRetry: cacheStats.retryStats.successAfterRetry,
+        finalFailures: cacheStats.retryStats.finalFailures,
+      },
+    });
+  }
```

Comment:
Not everyone might be interested in these logs. Let's keep this to debug mode for now. The logs can also add to the excess cost, as well as performance gets affected due to logs.